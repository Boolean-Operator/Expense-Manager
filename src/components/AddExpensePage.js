import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

//Add Expense Component
export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/')
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
        </h1>
      </div>    
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
   startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

  export default connect(undefined, mapDispatchToProps)(AddExpensePage);