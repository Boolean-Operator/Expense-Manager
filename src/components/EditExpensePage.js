import React from 'react';
import { connect } from 'react-redux';
import  ExpenseForm from './ExpenseForm'
import { editExpense, removeExpense } from '../actions/expenses'

// Edit Expense Component
export class EditExpensePage extends React.Component {
  
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onClick = () => {
    this.props.removeExpense({id:this.props.expense.id});
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Edit Expense
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button
          expense={this.props.expense}
          onClick={this.onClick}>Remove</button>
        </h1>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (data) => dispatch(removeExpense(data))
});
    
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);