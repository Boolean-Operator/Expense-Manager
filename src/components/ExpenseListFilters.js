import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount , sortByDate, setStartDate, setEndDate } from '../actions/filters'
import { DateRangePicker } from 'react-dates'

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate)),
    this.props.dispatch(setEndDate(endDate))
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };
  render() {
    return (
      <div>
        <input
          type='text'
          value={this.props.filters.text}
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value))
          }} 
        />
        <select 
          value={this.props.filters.sortBy}
          onChange={(e) => {
          (e.target.value === 'date') ?
          this.props.dispatch(sortByDate()) :
          this.props.dispatch(sortByAmount());
        }}>
          <option value='date'>Date</option>
          <option value='amount'>Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          // startDateId={}
          endDate={this.props.filters.endDate}
          // endDateId={}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}


        />
      </div>
    );

  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

export default connect(mapStateToProps)(ExpenseListFilters);