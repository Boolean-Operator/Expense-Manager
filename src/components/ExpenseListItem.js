import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

 //Expense List Item component

const ExpenseListItem = ({id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>${amount} -- Incurred:{moment(createdAt).format("MMM Do YYYY")}</p>
  </div>
);

export default ExpenseListItem;
