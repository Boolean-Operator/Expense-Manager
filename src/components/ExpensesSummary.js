// Expenses Summary Component

// Render in ExpenseDashboardPage

//Test with two snapshots
// Viewing 1 expense totalling $94.34 
// Viewing 2 expenses totalling $94.34

// Connect to store for:
  // expensesCount (of visible expenses)
  // expensesTotal (of visible expenses)

// Commit and Deploy to Heroku


import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal  = numeral(expensesTotal / 100).format('$0,0.00');
 
  return (
    <div>
      <h2>
        Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}
      </h2>
    </div>
  );
};
 
const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }

}

export default connect(mapStateToProps)(ExpensesSummary)

