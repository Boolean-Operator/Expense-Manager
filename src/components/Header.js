import React from 'react';
import { NavLink } from 'react-router-dom';

// Header Component
const Header = () => (
  <header>
    <h1>Expensify Header</h1>
    <h2>
      <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard-*-</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    </h2>
  </header>
)
export default Header;