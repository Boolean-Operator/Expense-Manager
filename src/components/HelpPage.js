import React from 'react';

const HelpPage = () => (
  <div>
    <h1>
      Help Page component
    </h1>
  </div>
);

export default HelpPage;