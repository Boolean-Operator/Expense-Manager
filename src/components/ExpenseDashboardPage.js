import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary';

// remove h1 tags
const ExpenseDashboardPage = () => (
  <div>
    <h1>
      <ExpensesSummary />
      <ExpenseListFilters />
      <ExpenseList />
    </h1>
  </div>
);

export default ExpenseDashboardPage; 