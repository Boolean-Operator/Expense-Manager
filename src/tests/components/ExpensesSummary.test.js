import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render Expenses Summary correctly with a single expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Expenses Summary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={231235425} />);
  expect(wrapper).toMatchSnapshot();
});