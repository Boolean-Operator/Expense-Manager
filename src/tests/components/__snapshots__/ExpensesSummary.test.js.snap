// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Summary correctly with a single expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h2>
</div>
`;

exports[`should render Expenses Summary correctly with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    23
     
    expenses
     totalling 
    $2,312,354.25
  </h2>
</div>
`;
